extension/keysocket-deezer.js

```diff
@@ -1,38 +1,23 @@
 function onKeyPress(key) {
-    if (document.getElementById('player_control_play')) { // old deezer style
+	if(document.querySelector('#player_control_play')) { // old deezer style
 		if(key === NEXT) {
-			var nextButton = document.getElementById('player_control_next');
-			simulateClick(nextButton);
+			simulateClick(document.querySelector('#player_control_next'));
 		} else if(key === PLAY) {
-			var isPlaying = document.getElementById('player_control_play').style.display === 'none';
-			var playPauseButton = null;
-			if(isPlaying) {
-			  playPauseButton = document.getElementById('player_control_pause');
-			} else {
-			  playPauseButton = document.getElementById('player_control_play');
-			}
-			simulateClick(playPauseButton);
+			simulateClick(document.querySelector('#player_control_play').style.display === 'none' ? document.querySelector('#player_control_pause') : document.querySelector('#player_control_play'));
```

Comment:
Whoa. Please separate into functions. Also you might benefit from a
switch() statement.

On Thu, Mar 5, 2015 at 11:57 AM, Eric Nemchik <notifications@github.com>
wrote:

> In extension/keysocket-deezer.js
> <https://github.com/borismus/keysocket/pull/99#discussion_r25896063>:
>
> >  		} else if(key === PLAY) {
> > -			var isPlaying = document.getElementById('player_control_play').style.display === 'none';
> > -			var playPauseButton = null;
> > -			if(isPlaying) {
> > -			  playPauseButton = document.getElementById('player_control_pause');
> > -			} else {
> > -			  playPauseButton = document.getElementById('player_control_play');
> > -			}
> > -			simulateClick(playPauseButton);
> > +			simulateClick(document.querySelector('#player_control_play').style.display === 'none' ? document.querySelector('#player_control_pause') : document.querySelector('#player_control_play'));
>
> how about something like this?
>
> function onKeyPress(key) {
>   var clickButton;
>   if(document.querySelector('#player_control_play')) { // old deezer style
>     if(key === NEXT) {
>       clickButton = document.querySelector('#player_control_next');
>     } else if(key === PLAY) {
>       if(document.querySelector('#player_control_play').style.display === 'none') {
>         clickButton = document.querySelector('#player_control_pause');
>       } else {
>         clickButton = document.querySelector('#player_control_pause');
>       }
>     } else if(key === PREV) {
>       clickButton = document.querySelector('#player_control_prev');
>     } else if(key === STOP) {
>       clickButton = document.querySelector('#player_control_pause');
>     }
>   } else { // new deezer style
>     if(key === NEXT) {
>       clickButton = document.querySelector('.control-next')[0);
>     } else if(key === PLAY) {
>       if(document.querySelector('.control-play')[0]) {
>         clickButton = document.querySelector('.control-play')[0];
>       } else {
>         clickButton = document.querySelector('.control-pause')[0];
>       }
>     } else if(key === PREV) {
>       clickButton = document.querySelector('.control-prev')[0];
>     } else if(key === STOP) {
>       clickButton = document.querySelector('.control-pause')[0];
>     }
>   }
>   simulateClick(clickButton);
> }
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/borismus/keysocket/pull/99/files#r25896063>.
>